### PR TITLE
Adds time gap detection to spike test

### PR DESCRIPTION
**Pull request by Bobfrat**
@lukecampbell Sets flag to Unknown if spike test flags data in a time gap.

### CODE REVIEW

File: ioos_qartod/qc_tests/qc.py [hunk 1]
```diff
@@ -177,6 +181,16 @@ def spike_check(arr, low_thresh, high_thresh, prev_qc=None):
     flag_arr = ((val < low_thresh) +
                 ((val >= low_thresh) & (val < high_thresh)) * QCFlags.SUSPECT +
                 (val >= high_thresh) * QCFlags.BAD_DATA)
+
+    # Change flag to unknown for data around large time gaps that failed the spike test
+    dt = np.mean(np.diff(times))  # mean value time between data samples
```
Comment (lukecampbell) [inline ioos_qartod/qc_tests/qc.py]: Could this be something more forgiving than the mean, like 1σ

Comment (Bobfrat) [inline ioos_qartod/qc_tests/qc.py]: For sure, I'll test that out

Comment (lukecampbell) [inline ioos_qartod/qc_tests/qc.py]: honestly, I don't even know if 1σ is the right answer here either. 

obs-data

timediffs

**Comment by ocefpaf** [comment]
Closing and re-opening just to get Travis-CI going. I just enabled it here.

**Comment by ocefpaf** [comment]
We need to merge https://github.com/ioos/qartod/pull/13 before this one to get the tests to run.

File: ioos_qartod/qc_tests/qc.py [hunk 2]
```diff
@@ -183,7 +183,7 @@ def spike_check(times, arr, low_thresh, high_thresh, prev_qc=None):
                 (val >= high_thresh) * QCFlags.BAD_DATA)
 
     # Change flag to unknown for data around large time gaps that failed the spike test
-    dt = np.mean(np.diff(times))  # mean value time between data samples
+    dt = np.percentile(np.diff(times), 75)  # mean value time between data samples
```
Comment (ocefpaf) [inline ioos_qartod/qc_tests/qc.py]: Please fix the comment too :wink:

**Comment by lukecampbell** [comment]
The problem with using statistical functions is that if the stream is completely contiguous and the distribution of time differences is really tight we'll be potentially cutting off arbitrary test points because any time delays in the 75th percentile will fall into this. 

I really don't know a good approach here.

**Pull request merged**